src/app/services/arduino-uploader/protocols/avrdude/index.ts

```diff
@@ -4,6 +4,45 @@ import BaseProtocol from "../base";
 export default class Avrdude extends BaseProtocol {
 
     async upload(response: Record<string, string>) {
-        throw new Error("Not implemented")
+        const Module = await import('@leaphy-robotics/avrdude-webassembly/avrdude.js')
+        const avrdude = await Module.default()
+        window["funcs"] = avrdude;
+        // check if port is open
+        if (this.port.readable || this.port.writable) {
+            await this.port.close()
+        }
+        await this.port.open({ baudRate: 115200 })
+        window["activePort"] = this.port
+        const avrdudeConfig = await fetch('/avrdude.conf').then(res => res.text())
+        avrdude.FS.writeFile('/tmp/avrdude.conf', avrdudeConfig)
+        avrdude.FS.writeFile('/tmp/program.hex', response['hex'])
+        let args = '';
+        // get board
+        const mcu = this.uploader.appState.getSelectedRobotType().micrcontroller
+        if (mcu == 'atmega328p') {
+            args = 'avrdude -P /dev/null -V -v -p atmega328p -c stk500v1 -C /tmp/avrdude.conf -b 115200 -D -U flash:w:/tmp/program.hex:i'
```

Comment:
the inclusion of metadata could also include the programmer, removing the need for if statements here